smarts/core/agent_interface.py

```diff
@@ -129,6 +129,23 @@ class AgentType(IntEnum):
     """Agent performs trajectory tracking using model predictive control."""
 
 
+@dataclass(frozen=True)
+class AgentsAliveDoneCriteria:
+    total_ego_agents_alive: Optional[int] = None
+    """if set, trigggers the agent to be done if total number of alive ego agents falls below total_agents_alive"""
+    total_social_agents_alive: Optional[int] = None
+    """if set, trigggers the agent to be done if total number of alive social agents falls below total_social_agents_alive"""
+    agent_lists_alive: Optional[List[Tuple[List[str], int]]] = None
+    """a list of agent lists to check alive or not, structure is List[Tuple[list_of_agent_ids, minimum_number_of_agents_alive in the list]]
```

Comment:
nit: I think perhaps that the tuple could be replaced with a class to allow for a descriptive condition.

Reply:
Sounds good! will do